sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TypeUtils.scala

```diff
@@ -73,7 +73,7 @@ object TypeUtils {
   }
 
   def compareBinary(x: Array[Byte], y: Array[Byte]): Int = {
-    val limit = math.min(x.length, y.length)
+    val limit = if (x.length <= y.length) x.length else y.length
```

Comment:
no need to use val keep the array length
`val al = a.length; val bl = b.length; val limit = if (al <= bl) al else bl` convert to 13 instructions
`if (x.length <= y.length) x.length else y.length` convert to 11 instructions

Reply:
How about just a `min`?